src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -38,32 +29,42 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             await foreach (var resourceEvent in resourceEvents.ConfigureAwait(false))
             {
                 var resourceName = resourceEvent.Resource.Name;
+                ResourceMonitorState? state;
 
-                _latestEvents[resourceName] = resourceEvent;
+                lock (_resourceMonitoringStates)
```

Comment:
This code isn't multi-threaded, right? There is just 1 thread/loop processing these events. So is there a need to lock here?

Reply:
I guess there is an internal `GetResourceMonitorState` that is used for tests that could be on a different thread.

Reply:
Yes. I didn't want tests to fail because the dictionary was in a bad state during a get.

The lock will never be contended in the real world, and this isn't performance critical code, so it seemed better to add some safety in the cause of test stability.